### PR TITLE
CP-15408: Add license restrictions

### PR DESCRIPTION
**Pull request by MihaelaStoica**
Port changes from cream-indigo for the additional restrictions

### CODE REVIEW

File: XenAdmin/Dialogs/VmSnapshotDialog.cs [hunk 1]
```diff
@@ -175,7 +175,7 @@ private void CheckpointInfoPictureBox_Click(object sender, EventArgs e)
         {
             string tt;
             if (Helpers.FeatureForbidden(_VM, Host.RestrictCheckpoint))
-                tt = string.Format(Messages.FEATURE_NOT_AVAILABLE_NEED_ENTERPRISE_OR_PLATINUM_PLURAL.Replace("\\n", "\n"), Messages.DISKMEMORY_SNAPSHOTS);
+                tt = Messages.FIELD_DISABLED;
```
Comment (cheng--zhang) [inline XenAdmin/Dialogs/VmSnapshotDialog.cs]: Can we add more information here like:
string.Format(Messages.FIELD_DISABLED.Replace("\\n", "\n"), Messages.DISKMEMORY_SNAPSHOTS);
So the detail reason can also go with the message.

**Comment by MihaelaStoica** [comment]
@cheng--zhang More information about the license type could be useful for the user, but we would need to make sure that the correct information is displayed for each XenServer version that XenCenter supports, as the license editions for which the feature is available can be different from one server version to the other. Given all this, I think the generic message is OK here (better than the wrong message, anyway)

File: XenAdmin/Commands/Controls/MigrateVMToolStripMenuItem.cs [hunk 2]
```diff
@@ -89,6 +89,10 @@ public MigrateVMCommand2(IMainWindow mainWindow, IEnumerable<SelectedItem> selec
 
             private static bool CanExecute(VM vm)
             {
+                if (vm.Connection != null && vm.Connection.IsConnected)
+                    if (Helpers.FeatureForbidden(vm.Connection, Host.RestrictInterPoolMigrate))
```
Comment (GaborApatiNagy) [inline XenAdmin/Commands/Controls/MigrateVMToolStripMenuItem.cs]: Could we use Intra- or Cross- instead of Inter- to have the same text everywhere?

**Comment by GaborApatiNagy** [comment]
Looks fine

**Pull request merged**